### ISSUE TITLE
Improve macro for checking conditions

### ISSUE THREAD

**Issue by basvodde** [cpputest/cpputest#8]
Currently, the macro CHECK(cond) is provided for checking arbitrary conditions.
A typical use is CHECK(actual>=minimum). Unfortunately, the output of this macro is somewhat limited.
For the example above, the failure message is "CHECK(actual>=minimum) failed."

To get more insight into the problem, I found the following macro very helpful:
# define CHECK_RELATION(first, relop, second)\

CHECK_LOCATION((first) relop (second), (StringFrom(#first #relop #second ", ") \
+StringFrom(first)+StringFrom(#relop)+StringFrom(second)).asCharString(), **FILE**, **LINE**)

Example: CHECK_RELATION(actual, >=, minimum)
Output on failure: "CHECK(actual>=minimum, 0.5>=0.8) failed"

In conclusion, I suggest to add the above macro (or something a bit more elegant) to utestmacros.h.

Also close in SourceForge at:
https://sourceforge.net/tracker/?func=detail&aid=3365842&group_id=199196&atid=968521
